### ISSUE TITLE
Restore session when it is empty after redirect

### ISSUE THREAD

**Issue by JanRogge** [pickware/ShopwareStripePayment#62]
There seems to be the possibility that when coming back to the `completeRedirectFlow` the session is empty which results in a failure to create the order.

To prevent this from happening an option is to restore the session if it is empty like we do for our webhooks. https://github.com/pickware/ShopwareStripePayment/blob/8cbf58f4bf60fed263ff2c950ee3096c20e49459/Subscriber/Frontend/Frontend.php#L41-L78

**Comment by ct-lh** [comment]
Hi,

Although this issue is almost 2 years old, I hope there is a chance this will be fixed/implemented.

We also have this issue from time to time, especially for customers which try to pay via iDeal via a mobile device (although I doubt that the customer's device is an issue here).

This is the error we could find in the server logs:

```
PHP Fatal error: Uncaught Error: Call to undefined method ShopwarePlugin\\PaymentMethods\\Components\\GenericPaymentMethod::getSnippet() in
/.../engine/Shopware/Plugins/Community/Frontend/StripePayment/Controllers/Frontend/StripePayment.php:114
Stack trace:
#0 /.../engine/Library/Enlight/Controller/Action.php(192): Shopware_Controllers_Frontend_StripePayment->completeRedirectFlowAction()
#1 /.../engine/Library/Enlight/Controller/Dispatcher/Default.php(478): Enlight_Controller_Action->dispatch('completeRedirec...')
#2 /.../engine/Library/Enlight/Controller/Front.php(228): Enlight_Controller_Dispatcher_Default->dispatch(Object(Enlight_Controller_Request_RequestHttp), Object(Enlight_Controller_Response_ResponseHttp))
#3 /.../engine/Shopware/Kernel.php(188): Enlight_Controller_Front->dispatch()
#4 /.../vendor/symfony/http-kernel/HttpCache/SubRequestHandler.php(102): Shopware\\Kernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#5 /.../vendor/symfony/ in /.../engine/Shopware/Plugins/Community/Frontend/StripePayment/Controllers/Frontend/StripePayment.php on line 114, referer: https://hooks.stripe.com/
```

From what I could retrace in `StripePayment/Controllers/Frontend/StripePayment.php:114` the cause for this is that the Stripe Session was lost in the shop.

The result of this error is that there are sometimes cases where the money is deducted from the customer's bank account but not transferred to us. It usually takes a few days for Stripe to refund it to the customer automatically.

Obviously, that is really customer unfriendly and we hope to find a fix for this.

From Stripe's API logs I could see that the request to their `/v1/sources` endpoint works like usual but the subsequent call of the `/v1/charges` endpoint (which is called in the case of payments without the error) is missing.

Best regards,
Lucas

**Comment by JanRogge** [comment]
Released with v5.4.2
